dataedit/static/dataedit/metadata.js

```diff
@@ -1,139 +1,119 @@
-    'language':0,
+function format_label(s){
+  // capitalize the string
+  if (typeof s !== 'string') return ''
+  s = s.split('_').pop()
+  return s.charAt(0).toUpperCase() + s.slice(1)
 };
 
-function create_box($parent){
-    var $container = $('<div class="metacontainer" id='+prefix+indexes[prefix]+'></div>');
 
-    var $header = $('<div class="metacontainer-header"></div>')
-
-    var $removeBtn = $('<a style="color:white" onclick="$(\'#'+prefix+indexes[prefix]+'\').remove();"><span class="glyphicon glyphicon-minus-sign"/></a>');
-    var $subcontainer = $('<div class="metaformframe" id='+prefix+indexes[prefix]+'></div>')
-
-    $header.append($removeBtn);
-    $container.append($header);
-    $container.append($subcontainer);
-    $parent.append($container);
-    return $subcontainer
-}
-
-function add_labeled_textfield($container, label, id, value){
-    var $input_label = $('<label for="' + id + '">'+label+'</label>');
-    var $input = $('<input autocomplete="off" class="input form-control" id="' + id + '" name="' + id + '" type="text" value="' + value + '">');
-    $container.append($input_label);
-    $container.append($input);
-}
-
-function add_language($parent, obj){
-    prefix='language';
-    indexes[prefix] = indexes[prefix] +1;
-    var $container = create_box($parent);
+function extract_el_index(name){
+    // get the number at the end of the name
+  	name = name.split('_').pop()
+	return name.match(/\d+/)[0]
+};
 
-    if('undefined' === typeof obj){
-        var obj={};
-        obj.language='';
-    }
 
-    add_labeled_textfield($container, 'Language', prefix+indexes[prefix], obj.language);
+function extract_el_name(name, idx){
+    // get the field at the end of the name, without the number
+  	name = name.substring(0, name.length - idx.length);
+	return name
 };
 
-function add_spatial($parent, obj){
-    prefix='spatial';
-    indexes[prefix] = indexes[prefix] +1;
-    var $container = create_box($parent);
 
-    if('undefined' === typeof obj){
-        var obj={};
-        obj.extent='';
-        obj.resolution='';
-    }
+function remove_element(parent){
+    var idx = extract_el_index(parent);
+    parent = extract_el_name(parent, idx);
 
-    add_labeled_textfield($container, 'Extent', prefix+indexes[prefix]+'_extent', obj.extent);
-    add_labeled_textfield($container, 'Resolution', prefix+indexes[prefix]+'_resolution', obj.resolution);
+    //find element container
+    var $parent = $('#' + parent + '_container');
+    // if element is not the last, proceed to removal
+    if($parent[0].childElementCount > 1){
+        var $container = $('#'+ parent + idx);
+        $container.remove();
+    }
 };
 
-function add_temporal($parent, obj){
-    prefix='temporal';
-    indexes[prefix] = indexes[prefix] +1;
-    var $container = create_box($parent);
 
-    if('undefined' === typeof obj){
-        var obj={};
-        obj.start='';
-        obj.end='';
-		obj.resolution='';
+function add_list_objects(prefix){
```

Comment:
this function replaces all the add_* functions, it copies an object from the list, re-initializes its values and forms and replace the ids of the html component with a new one